### ISSUE TITLE
Visual Recognition v3 API key lookup problems

### ISSUE THREAD

**Issue by l2fprod** [watson-developer-cloud/java-sdk#371]
In https://github.com/watson-developer-cloud/java-sdk/blob/master/src/main/java/com/ibm/watson/developer_cloud/visual_recognition/v3/VisualRecognition.java#L71

```
  private static final String SERVICE_NAME = "visual_recognition";
```

This is the old name for the Visual Recognition service. With V3 (the combined Watson + Alchemy), the service name in the VCAP_SERVICES is `watson_vision_combined`, thus the call to CredentialUtils.getAPIKey(name) in WatsonService.java finds no API key.

The node-sdk uses the right name for V3: https://github.com/watson-developer-cloud/node-sdk/blob/194fc77d76c647244033d856ef7c25d9c9722484/lib/index.js#L67 when doing this lookup.

I tried the following as a workaround:

```
vision = new VisualRecognition(VisualRecognition.VERSION_DATE_2016_05_19);
vision.setApiKey(CredentialUtils.getAPIKey("watson_vision_combined"));
```

but then it ends up in this code in CredentialUtils:

```
            if (serviceName.equalsIgnoreCase(ALCHEMY_API)) {
              return credentials.get(APIKEY).getAsString();
            } else {
              final String username = credentials.get(USERNAME).getAsString();
              final String password = credentials.get(PASSWORD).getAsString();
              return Credentials.basic(username, password);
            }
```

and tries to find username and password instead of api_key.

Am I missing something or is the SDK not ready for the new service? I'm using 3.0.1.

**Comment by antonatos** [comment]
It seems my issue is somehow related:

I am using the VisualRecognition API. My JDK version is 1.8 and the java-sdk version is 3.0.1: 
         <dependency>
            <groupId>com.ibm.watson.developer_cloud</groupId>
            <artifactId>java-sdk</artifactId>
            <version>3.0.1</version>
        </dependency>

I have created a sample application like this (most of it is copy-paste from the documentation):
       VisualRecognition service = new VisualRecognition(VisualRecognition.VERSION_DATE_2016_05_19);
        service.setApiKey("<my api key here>");
        System.out.println("Classify an image");
        ClassifyImagesOptions options = new ClassifyImagesOptions.Builder()
                .images(new File("src/test/resources/golf.png"))
                .build();
        VisualClassification result = service.classify(options).execute();
        System.out.println(result);
    }

The golf.png loaded is downloaded from this location: https://www.goultralow.com/wp-content/uploads/2015/02/Golf-GTE-for-GUL2.png

If I try the demo web app, the image classification works correctly. When trying from my application I get empty results. The weird behavior also persists if I use an invalid key. This probably prompts to obsolete endpoints for verification and classification service.

**Comment by jsstylos** [comment]
I don't think the VCAP_SERVICES part of this issue was ever fixed.

**Comment by germanattanasio** [comment]
Fixed in #693
